modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -187,24 +194,87 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
+    // classloader reference, it has to be extracted manually though reflection from each delegate/wrapper class.
+    // Hierarchy leading to real ThreadPoolExecutor is: AbandonedConnectionCleanupThread.cleanupThreadExecutorService ->
+    // class DelegatedExecutorService.e -> class ThreadPoolExecutor.
+    // Note that the field 'cleanupThreadExcecutorService' is mispelled. There's actually a typo in MySql driver code.
+    try {
+      Field cleanupExecutorServiceField = cleanupThreadsClass
+          .getDeclaredField("cleanupThreadExcecutorService");
+      cleanupExecutorServiceField.setAccessible(true);
+      ExecutorService delegateCleanupExecutorService =
+          (ExecutorService) cleanupExecutorServiceField.get(cleanupThreadsClass);
+
+      Field realExecutorServiceField = delegateCleanupExecutorService.getClass().getSuperclass().getDeclaredField("e");
+      realExecutorServiceField.setAccessible(true);
+      ThreadPoolExecutor realExecutorService =
+          (ThreadPoolExecutor) realExecutorServiceField.get(delegateCleanupExecutorService);
+
+      // Set cleanup thread executor service thread factory to one whose classloader is the system one
+      realExecutorService.setThreadFactory(Executors.defaultThreadFactory());
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      logger.warn("Error cleaning threadFactory from AbandonedConnectionCleanupThread executor service", e);
+    }
+  }
+
+  /**
+   * Sends a shutdown message to MySql's connection cleanup thread class.
+   * 
+   * @param classAbandonedConnectionCleanupThread
+   * @throws IllegalAccessException
+   * @throws InvocationTargetException
+   * @throws NoSuchMethodException
+   */
+  private void shutdownMySqlConnectionCleanupThreads(Class<?> classAbandonedConnectionCleanupThread)
+      throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+    try {
+      Method uncheckedShutdown = classAbandonedConnectionCleanupThread.getMethod("uncheckedShutdown");
+      uncheckedShutdown.invoke(null);
+    } catch (NoSuchMethodException e) {
+      Method checkedShutdown = classAbandonedConnectionCleanupThread.getMethod("shutdown");
+      checkedShutdown.invoke(null);
+    }
+  }
+
+  /**
+   * Tries to find the MySql driver AbandonedConnectionCleanupThread class, with the known class addresses.
+   * 
+   * @return The MySql driver AbandonedConnectionCleanupThread class object, if found.
+   */
+  private Class<?> findMySqlDriverClass() throws ClassNotFoundException {
+    Class<?> foundClass = null;
+    for (String knownCleanupThreadClassAddress : CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES) {
+      try {
+        return this.getClass().getClassLoader().loadClass(knownCleanupThreadClassAddress);
+      } catch (ClassNotFoundException e) {
+        logger.warn("No AbandonedConnectionCleanupThread registered with class address " + knownCleanupThreadClassAddress);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Format specifiers should be used instead of string concatenation. ([squid:S3457](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3457))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAu&open=AWjzLS16otIXnmsUYkAu)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAu